### ISSUE TITLE
Best practice for "generic" Jimp instance type missing from docs?

### ISSUE THREAD

**Issue by LostInStatic** [jimp-dev/jimp#1328]
**Is your feature request related to a problem? Please describe.**
I'm working on a solution that will load an image from url or a buffer, transform it and output it in base64. This means it will be passed around through a couple functions. I've tried to figure out a type for the Jimp object being passed as parameter, but nothing I tried works (see examples below). Resources from the internet show examples with older versions of Jimp (with default import). 

AFAIK it was possible with older versions just by doing this:
```typescript
import Jimp from "jimp"

function transform(jimpImage: Jimp) {
//...
}
```

 Is it not possible anymore and I should treat url and buffer inputs as completely separate pipelines?

**Describe the solution you'd like**
Hopefully it's just a case of docs not being clear?

**Additional context**

[Sandbox](https://codesandbox.io/p/sandbox/pc26m9)
```typescript
import { Jimp } from "jimp";

// 'Jimp' refers to a value, but is being used as a type here.
async function transform(jimpImage: Jimp) {
  return await jimpImage.getBase64("image/png");
}

// Property 'getBase64' does not exist
async function transform1(jimpImage: typeof Jimp) {
  return await jimpImage.getBase64("image/png");
}

//Using just one return type will not accept the output of the other function
//Results in:
// This expression is not callable. Each member of the union type [...] has signatures, but none of those signatures are compatible with each other.
async function transform2(
  jimpImage:
    | Awaited<ReturnType<typeof Jimp.read>>
    | Awaited<ReturnType<typeof Jimp.fromBuffer>>
) {
  return await jimpImage.getBase64("image/png");
}
```

**Comment by LostInStatic** [comment]
Haha, now I feel stupid - did not think about looking for `InstanceType` in TS docs even though this is literally what I've been thinking of. Thanks a lot!

**Comment by hipstersmoothie** [comment]
will be released soon!

**Comment by hipstersmoothie** [comment]
:rocket: Issue was released in [`v1.4.0`](https://github.com/jimp-dev/jimp/releases/tag/v1.4.0) :rocket:

**Comment by vntw** [comment]
Hey @hipstersmoothie, thanks for all the improvements and new release 🚀 

I upgraded to v1 without major issues but one question regarding the instance type: Should `new Jimp(…)` and `await Jimp.read` yield the same type? I used to pass a Jimp instance around interchangeably, which errors now. Hopefully this [mini example](https://codesandbox.io/p/sandbox/pk8qdr?file=%2Findex.mts%3A11%2C1) shows what I mean (I also included the snippet of this issue in a separate file)

The [docs](http://jimp-dev.github.io/jimp/api/jimp/classes/jimp/#read) still say that `Jimp.read` returns a Jimp instance - maybe a different one or just a type error?

Would be great to hear your thoughts on this, thanks! 🙂
